test/e2e/kubectl.go

```diff
@@ -1201,3 +1200,34 @@ func streamingUpload(file *os.File, fileName string, postBodyWriter *multipart.W
 		Failf("Unable to close the writer for file upload. Error: %s", err)
 	}
 }
+
+var binPrefixes = []string{
+	"_output/dockerized/bin",
+	"_output/local/bin",
+	"platforms",
+}
+
+// findBinary searches through likely paths to find the specified binary.  It
+// takes the one that has been built most recently.  Platform should be
+// specified as '<os>/<arch>'.  For example: 'linux/amd64'.
+func findBinary(binName string, platform string) (string, error) {
+	var binTime time.Time
+	var binPath string
+
+	for _, pre := range binPrefixes {
+		tryPath := path.Join(testContext.RepoRoot, pre, platform, binName)
+		fi, err := os.Stat(tryPath)
+		if err != nil {
+			continue
+		}
+		if fi.ModTime().After(binTime) {
```

Comment:
nit: It would be nice to log if a version was chosen because of a later build. I could see some confusion as to why one binary was chosen over another until reading the code.